### PR TITLE
Base provider cleanup

### PR DESCRIPTION
**Pull request by joan41868**
Signed-off-by: Yoan Sredkov <yoansredkov@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/tests/src.ts/test-providers.spec.ts [hunk 1]
```diff
@@ -882,19 +882,97 @@ testFunctions.push({
...
-        });
-        assert.ok(!!value);
-    });
-});
+// describe("Test Provider Methods", function() {
```
Comment (Daniel-K-Ivanov) [inline packages/tests/src.ts/test-providers.spec.ts:885]: We can remove the commented tests here and add `TODOs` for adding tests for the provider methods once we implement them

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]
Good progress towards cleaning the provider classes!

File: packages/providers/src.ts/base-provider.ts [hunk 2]
```diff
@@ -30,39 +28,6 @@ import axios from "axios";
...
-            const sorted = Object.keys(unique);
-            sorted.sort();
-
-            return sorted.join("|");
-
-        } else {
-            return checkTopic(topic);
-        }
-    }).join("&");
-}
```
[the remark this answers is not in the document]
Comment (joan41868) [reply packages/providers/src.ts/base-provider.ts:65]: I will leave events and the serdes for now, as I found usage in the `contracts` package.

File: packages/providers/src.ts/base-provider.ts [hunk 3]
```diff
@@ -707,131 +504,10 @@ export class BaseProvider extends Provider implements EnsProvider {
...
-        this._lastBlockNumber = blockNumber;
-
-        // Once all events for this loop have been processed, emit "didPoll"
-        Promise.all(runners).then(() => {
-            this.emit("didPoll", pollId);
-        }).catch((error) => { this.emit("error", error); });
-
-        return;
-    }
+    async poll(): Promise<void> {}
```
Comment (Daniel-K-Ivanov) [inline packages/providers/src.ts/base-provider.ts]: Do we need to keep this function? I think that we do not need it

File: packages/providers/src.ts/base-provider.ts [hunk 4]
```diff
@@ -707,131 +504,10 @@ export class BaseProvider extends Provider implements EnsProvider {
...
 
     // Deprecated; do not use this
     resetEventsBlock(blockNumber: number): void {
```
Comment (Daniel-K-Ivanov) [inline packages/providers/src.ts/base-provider.ts]: Can't we delete this function. It seems that we wont need it

File: packages/providers/src.ts/base-provider.ts [hunk 5]
```diff
@@ -1113,38 +728,8 @@ export class BaseProvider extends Provider implements EnsProvider {
...
-            return hexlify(result);
-        } catch (error) {
-            return logger.throwError("bad result from backend", Logger.errors.SERVER_ERROR, {
-                method: "call",
-                params, result, error
-            });
-        }
-    }
-
     async estimateGas(transaction: Deferrable<TransactionRequest>): Promise<BigNumber> {
```
Comment (Daniel-K-Ivanov) [inline packages/providers/src.ts/base-provider.ts:692]: Instead of returning `0`, lets throw not implemented/not supported

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

**Review by Daniel-K-Ivanov — APPROVED** [review]
Lets wait for CICD to go through and we can merge

**Pull request merged**